### PR TITLE
Add `description` kwarg to command.Parameter and show in help command

### PR DESCRIPTION
**Pull request by Soheab**
## Summary

This PR adds a way to set a description for an ext.commands/hybrid command argument.
and reformats the `DefaultHelpCommand` to show that description, MinimalHelpCommand is left unchanged.

Also added `arguments_heading` and `default_argument_description` keyword arguments to `DefaultHelpCommand`.

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/ext/commands/parameters.py [hunk 1]
```diff
@@ -152,6 +158,11 @@ def converter(self) -> Any:
 
         return self.annotation
 
+    @property
+    def description(self) -> Optional[str]:
+        """Optional[:class:`str`]: The description of this parameter."""
+        return self._description if self._description is not empty else None
```
Comment (Gobot1234) [inline discord/ext/commands/parameters.py:164]: This shouldn't be empty right?

```suggestion
        return self._description if self._description is not MISSING else None
```

Comment (Gobot1234) [reply discord/ext/commands/parameters.py:164]: Sorry please ignore this

**Review by lol219 — APPROVED** [review]

File: discord/ext/commands/help.py [hunk 2]
```diff
@@ -1088,9 +1087,60 @@ def add_indented_commands(
...
-            entry = f'{self.indent * " "}{name:<{width}} {command.short_doc}'
+            entry = f'{self.indent * " "}{name:<{width}} - {command.short_doc}'
             self.paginator.add_line(self.shorten_text(entry))
 
+    def add_command_arguments(
+        self, command: Command[Any, ..., Any], /, *, heading: str, max_size: Optional[int] = None
+    ) -> None:
+        """Indents a list of command arguments after the specified heading.
+
+        The formatting is added to the :attr:`paginator`.
```
Comment (Rapptz) [inline discord/ext/commands/help.py:1098]:
```suggestion
```

File: discord/ext/commands/help.py [hunk 3]
```diff
@@ -1005,9 +992,19 @@ class DefaultHelpCommand(HelpCommand):
...
+
+        .. versionadded:: 2.0
     commands_heading: :class:`str`
         The command list's heading string used when the help command is invoked with a category name.
         Useful for i18n. Defaults to ``"Commands:"``
+    default_argument_description: :class:`str`
+        The default argument description string used when the argument's :attr:`~.commands.Parameter.description` is ``None``.
+        Useful for i18n. Defaults to ``"No description given."``
+
+        .. versionadded:: 2.0
```
Comment (Rapptz) [inline discord/ext/commands/help.py:1007]: Surely if no description is given then it should just be empty?

File: discord/ext/commands/help.py [hunk 4]
```diff
@@ -1088,9 +1087,60 @@ def add_indented_commands(
         for command in commands:
             name = command.name
             width = max_size - (get_width(name) - len(name))
-            entry = f'{self.indent * " "}{name:<{width}} {command.short_doc}'
+            entry = f'{self.indent * " "}{name:<{width}} - {command.short_doc}'
             self.paginator.add_line(self.shorten_text(entry))
 
+    def add_command_arguments(
+        self, command: Command[Any, ..., Any], /, *, heading: str, max_size: Optional[int] = None
```
Comment (Rapptz) [inline discord/ext/commands/help.py:1094]:
```suggestion
        self, command: Command[Any, ..., Any], /
```

This interface doesn't make much sense. The heading is already an attribute and the `max_size` doesn't need to be specified.

**Review by Rapptz — CHANGES_REQUESTED** [review]
I feel like there should probably be a way to disable this argument heading thing for those who don't want it.

File: discord/ext/commands/help.py (hunk 3 above)
Comment (Soheab) [reply discord/ext/commands/help.py:1007]: Correct but then the default (if any) looks out of place.
```
  name (default: ...)
  name - Description 
  name - Description (default: ...)
  name (default: ...)
```

File: discord/ext/commands/help.py (hunk 4 above)
Comment (Soheab) [reply discord/ext/commands/help.py:1094]: Ah my bad, that was copied from the signature of `DefaultHelpCommand.add_indented_commands`
https://github.com/Rapptz/discord.py/blob/584c22f642cfb63b921467a901c8a47c721923b3/discord/ext/commands/help.py#L1051-L1053

Comment (Rapptz) [reply discord/ext/commands/help.py:1094]: For that one, the heading used is dynamic so it makes a bit more sense. The documentation nit is still relevant there though...

File: discord/ext/commands/help.py [hunk 5]
```diff
@@ -1049,6 +1067,34 @@ def get_ending_note(self) -> str:
...
+        ------------
+        command: :class:`Command`
+            The command to get the signature of.
+
+        Returns
+        --------
+        :class:`str`
+            The signature for the command.
+        """
+        if self.show_parameter_descriptions is False:
```
Comment (Rapptz) [inline discord/ext/commands/help.py:1088]:
```suggestion
        if not self.show_parameter_descriptions:
```

File: discord/ext/commands/help.py [hunk 6]
```diff
@@ -1124,6 +1208,9 @@ def add_command_formatting(self, command: Command[Any, ..., Any], /) -> None:
                     self.paginator.add_line(line)
                 self.paginator.add_line()
 
+        if self.show_parameter_descriptions is True:
```
Comment (Rapptz) [inline discord/ext/commands/help.py:1211]:
```suggestion
        if self.show_parameter_descriptions:
```

**Review by Rapptz — APPROVED** [review]
Thanks.

**Pull request merged**